Engine/Engine.cs

```diff
@@ -245,11 +245,15 @@ public void Run(AlgorithmNodePacket job, string assemblyPath)
                         // Algorithm runtime error:
                         if (algorithm.RunTimeError != null)
                         {
-                            throw algorithm.RunTimeError;
+                            throw new Exception("Algorithm.RunTimeError", algorithm.RunTimeError); //Keep original exception as inner exception to preserve stack trace
                         }
                     }
                     catch (Exception err)
                     {
+                        while (err != null && err.InnerException != null)
+                        {
+                            err = err.InnerException; //Find inner most exception as this will be most useful to algorithm designer
```

Comment:
The logging should print out the entire stack, including inner exceptions, do it may be best to pass the full exception instance and let the logging component decide how to handle it